### PR TITLE
Add String Extensions

### PR DESCRIPTION
**Pull request by sebastianvarela**
Add String Extensions

### CODE REVIEW

**Comment by sebastianvarela** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Big PR, try to keep changes smaller if you can 😜</td>
    </tr>
  </tbody>
</table>
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          6 Messages
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false"><strong>📢 [iOS]</strong> Executed 44 tests, with 0 failures (0 unexpected) in 0.596 (0.700) seconds</td>
    </tr>
    <tr>
      <td>:book:</td>
      <td data-sticky="false"><strong>📢 [mac]</strong> Executed 43 tests, with 0 failures (0 unexpected) in 1.023 (1.149) seconds</td>
    </tr>
    <tr>
      <td>:book:</td>
      <td data-sticky="false"><strong>📢 [tv]</strong> Executed 43 tests, with 0 failures (0 unexpected) in 0.651 (0.710) seconds</td>
    </tr>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">⌛️ <strong>[iOS]</strong> Slowest test: test_html_value() took 0.496 seconds</td>
    </tr>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">⌛️ <strong>[mac]</strong> Slowest test: test_append_if_not_exists() took 0.490 seconds</td>
    </tr>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">⌛️ <strong>[tv]</strong> Slowest test: test_html_value() took 0.443 seconds</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: Source/Extensions/Foundation/StringExtensions+UIKit.swift [hunk 1]
```diff
@@ -0,0 +1,66 @@
+import Foundation
+import UIKit
+
+public extension String {
+//    var underLined: NSAttributedString {
```
Comment (r-pedraza) [inline Source/Extensions/Foundation/StringExtensions+UIKit.swift]: Commented code?

File: Tests/Extensions/Foundation/StringExtensions+ValidatorsTests.swift [hunk 2]
```diff
@@ -0,0 +1,41 @@
+import XCTest
+import Foundation
+import MagicPills
+
+class StringExtensionsValidatorsTests: XCTestCase {
```
Comment (r-pedraza) [inline Tests/Extensions/Foundation/StringExtensions+ValidatorsTests.swift]: I would separate test, isValidEmail and isNotValidEmail.

File: Tests/Extensions/Foundation/StringExtensions+ValidatorsTests.swift [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    func test_is_valid_email() {
+        XCTAssertTrue("wawa@wewe.com".isValidEmail)
+        XCTAssertTrue("hola@bq.com".isValidEmail)
+        XCTAssertFalse("chau".isValidEmail)
+    }
+
```
Comment (r-pedraza) [inline Tests/Extensions/Foundation/StringExtensions+ValidatorsTests.swift]: DITO

File: Tests/Extensions/Foundation/StringExtensions+ValidatorsTests.swift [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+    func test_is_valid_phone() {
+        XCTAssertTrue("687687687".isValidPhone)
+        XCTAssertTrue("787687687".isValidPhone)
+        XCTAssertTrue("987687687".isValidPhone)
+        XCTAssertFalse("587687687".isValidPhone)
+        XCTAssertFalse("124".isValidPhone)
+        XCTAssertFalse("kkk".isValidPhone)
+    }
+
+    func test_is_valid_nif() {
```
Comment (r-pedraza) [inline Tests/Extensions/Foundation/StringExtensions+ValidatorsTests.swift]: DITO

File: Tests/Extensions/Foundation/StringExtensions+ValidatorsTests.swift [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+        XCTAssertTrue("11234567X".isValidNIF)
+        XCTAssertTrue("21234567R".isValidNIF)
+        XCTAssertTrue("12345678Z".isValidNIF)
+        XCTAssertTrue("05446767E".isValidNIF)
+        XCTAssertFalse("05446767F".isValidNIF)
+        XCTAssertFalse("kkk".isValidNIF)
+        XCTAssertFalse("".isValidNIF)
+    }
+
+    func test_is_valid_nie() {
```
Comment (r-pedraza) [inline Tests/Extensions/Foundation/StringExtensions+ValidatorsTests.swift]: DITO

**Pull request merged**